From 498401d5d513e1ad3832eae7353584b9bc76b627 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Thu, 1 Aug 2013 10:02:04 +0800
Subject: libmediaplayerservice: fix hls can not pause/play issue in HTML5 fullscreen mode

BZ: 107204

fix hls can not pause/play issue in HTML5 fullscreen mode which is introduced in Android 4.3

Category: aosp improvement
Domain: Video.Media-streaming
Origin: internal
Upstream-Candidate: yes

Change-Id: I1452528e09abfb71959deb90688f44e92e68e445
Orig-Change-Id: I165e561a01faee95b3db599f13c397122b359dcd
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 media/libmediaplayerservice/nuplayer/NuPlayer.cpp |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
index c277fee..fb2a0b6 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
@@ -1466,6 +1466,13 @@ void NuPlayer::onSourceNotify(const sp<AMessage> &msg) {
                 schedulePollDuration();
             }
 
+            if (mDriver != NULL) {
+                sp<NuPlayerDriver> driver = mDriver.promote();
+                if (driver != NULL) {
+                    driver->notifyFlagsChanged(flags);
+                }
+            }
+
             mSourceFlags = flags;
             break;
         }
-- 
1.7.4.1

